doc/vhpidirect/examples/arrays.rst

```diff
@@ -0,0 +1,63 @@
+.. program:: ghdl
+.. _COSIM:VHPIDIRECT:Examples:arrays:
+
+Arrays
+#######
+
+Constrained/bounded integer arrays
+**********************************
+
+As explained in :ref:`Restrictions_on_foreign_declarations`, unconstrained arrays are represented by a fat pointer in C,
+and it is not suggested to use fat pointers unless it is unavoidable. Hence, it is desirable for data buffers which are
+to be represented as arrays to be constrained. However, constraining arrays in VHDL and C separatedly is error prone.
+
+This example includes multiple solutions to set the bounding size once only (either in C or in VHDL), and have the
+parameter passed to the other language so that matching types are defined. Moreover, independently of where is the size
+defined, it is possible to allocate and free the buffers in any of the languages. The following examples show how to
+define them in C or VHDL and have them (de)allocated in either of them.
+
+.. ATTENTION::
+  Pointers/accesses MUST be freed/deallocated in the same language that was used to allocate them. Hence, it is not
+  possible to allocate in VHDL and free in C or *vice versa*.
+
+:cosimtree:`Sized in C <vhpidirect/arrays/intvector/csized>`
+------------------------------------------------------------
+
+Integer arrays fully defined in C can be passed to VHDL by first passing their size, so that an appropriate array
+type can be created in VHDL to hold the array's access (pointer). After that, another VHPIDIRECT subprogram can be
+defined to return the array access.
+
+This example shows how to hardcode both the length and the initial content of an array in C. Matching types are created in
+VHDL, the pointer is passed and the content is read and modified from VHDL.
+
+If the integer array must be created or filled at runtime by some more advanced process, it is possible to execute the GHDL
+simulation within a custom ``main()`` entrypoint (see :ref:`COSIM:VHPIDIRECT:Examples:wrapping:basic`). This use case is
+included in the example too. By using ``main.c`` instead of ``caux.c``, the content of the array is written programatically
```

Comment:
Very neat. I was under the impression to use both files. This is more befitting.

Reply:
In fact, I tried, but it seems not to be possible to overwrite the hardcoded bounded array with the pointer returned by malloc. I could replace the pointer, but not the value returned by sizeof. Hence, a different variable needs to be used for the size. At that point, I felt that keeping them separate was better.

Reply:
Yeah, that's why my examples also ran tangential